tools/local/main.go

```diff
@@ -0,0 +1,347 @@
+	"k8sgateway/test/fake"
+)
+
+const (
+	network               = "acme"
+	gatewayPort           = 8443
+	gatewayHost           = "127.0.0.1"
+	kubeConfigPath        = "/Users/<username>/.kube/config-twingate-gateway-local"
+	clusterName           = "gateway-local-development"
+	kindClusterName       = "kind-" + clusterName
+	kindPort              = 6443
+	controllerPort        = 8080
+	kubeConfigClusterName = "twingate-gateway-local"
+	defaultUsername       = "alex@acme.com"
+)
+
+var kindClusterYaml = fmt.Sprintf(`
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  apiServerPort: %d
+nodes:
+- role: control-plane
+  extraMounts:
+    - hostPath: ./test/data/api_server/tls.crt
+      containerPath: /etc/kubernetes/pki/ca.crt
+    - hostPath: ./test/data/api_server/tls.key
+      containerPath: /etc/kubernetes/pki/ca.key
+`, kindPort)
+
+// Before running this local dev client, you need to do the following:
+// - Caddy must be already running. Run `caddy run` to start Caddy.
+// - Update `kubeConfigPath` above to point to your local KubeConfig file.
+func main() {
+	// Parse command line flags
+	username := flag.String("username", defaultUsername, "Username to use for authentication")
+	createKubeConfig := flag.Bool("create-kubeconfig", true, "Whether to create kubeConfig. If kubeConfig already exists, it will be overwritten")
+	flag.Parse()
+
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		log.Fatalf("Failed to create logger: %v", err)
+	}
+
+	defer func() {
+		_ = logger.Sync()
+	}()
+
+	zap.ReplaceGlobals(logger)
+
+	// Create a context that's canceled when SIGINT or SIGTERM is received
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		sig := <-sigs
+		logger.Info("Received signal, shutting down", zap.String("signal", sig.String()))
+		cancel()
+	}()
+
+	if err := createKindCluster(logger, *username); err != nil {
+		logger.Fatal("Failed to create kind cluster", zap.Error(err))
+	}
+
+	controller := fake.NewController(network, controllerPort)
+	defer controller.Close()
+
+	logger.Info("Controller is serving at", zap.String("url", controller.URL))
+
+	user := &token.User{
+		ID:       "user-1",
+		Username: *username,
+		Groups:   []string{"Developer", "OnCall"},
+	}
+
+	client := fake.NewClient(
+		user,
+		fmt.Sprintf("%s:%d", gatewayHost, gatewayPort),
+		controller.URL,
+		"https://127.0.0.1:6443",
+	)
+	defer client.Close()
+
+	logger.Info("Client is serving at", zap.String("url", client.URL))
+
+	if *createKubeConfig {
+		if err := createKubeConfigFile(client.URL); err != nil {
+			logger.Error("Failed to create kubeConfig", zap.Error(err))
+
+			return
+		}
+
+		logger.Info("Created kubeConfig", zap.String("path", kubeConfigPath))
+		logger.Info("You can now use kubectl with this config:",
+			zap.String("command", fmt.Sprintf("kubectl --kubeconfig=%s get pods", kubeConfigPath)))
```

Comment:
This log is unnecessary. If we want to provide an example how to use `kubectl` with this context, it should be in the instruction text at line 137 below.